src/TransferList/TransferList.stories.tsx

```diff
@@ -1,50 +1,152 @@
+      { id: "Grapes", primaryLabel: "Grapes" },
+      { id: "Cherry", primaryLabel: "Cherry" }
+    ]
+  },
+
+  render: TransferList
+};
+
+// use an array of strings with pre-transferred items
+export const WithStringArray: StoryObj<typeof TransferList<TransferListItem>> =
+  {
+    args: {
+      ...defaultArgs,
+      items: [
+        "Apples",
+        "Pears",
+        "Oranges",
+        "Bananas",
+        "Mangoes",
+        "Kiwi",
+        "Dragonfruit",
+        "Plum",
+        "Grapes",
+        "Cherry"
+      ],
+      targetListKeys: ["Apples", "Grapes"]
+    },
+    render: TransferList
   };
 
-  return (
-    <Box sx={{ height: 200 }}>
-      <TransferList
-        {...args}
-        selectedItems={selectedItems}
-        onChange={onChange}
-      />
-    </Box>
-  );
+// use a custom object shape array with filterKey and itemLabel configurations
+export const WithCustomObjectArray: StoryObj<
+  typeof TransferList<CustomTransferListItemType>
+> = {
+  args: {
+    ...defaultArgs,
+    filterKey: item => item.key,
+    itemProps: {
+      primaryLabel: item => item.name
+    },
+    items: [
+      { key: "A", name: "Apples" },
+      { key: "P", name: "Pears" },
+      { key: "O", name: "Oranges" },
+      { key: "B", name: "Bananas" },
+      { key: "M", name: "Mangoes" },
+      { key: "K", name: "Kiwi" },
+      { key: "D", name: "Dragonfruit" },
+      { key: "Pl", name: "Plum" },
+      { key: "G", name: "Grapes" },
+      { key: "C", name: "Cherry" }
+    ],
+    sourceListLabel: "Source List Label",
+    targetListKeys: ["C", "K"],
+    targetListLabel: "Target List Label"
+  },
+
+  render: TransferList
 };
 
-export const Default = {
+// use a pre-defined object shape sarry with no additional confiuration
+export const WithSecondaryLabel: StoryObj<
+  typeof TransferList<TransferListItem>
+> = {
   args: {
-    items: ["Apples", "Pears", "Oranges", "Banana", "Mangoes"],
-    onChange: () => {},
-    selectedItems: ["Apples"]
+    ...defaultArgs,
+    items: [
+      { id: "Apples", primaryLabel: "Apples", secondaryLabel: "Granny Smith" },
+      { id: "Pears", primaryLabel: "Pears", secondaryLabel: "Conference" },
+      { id: "Oranges", primaryLabel: "Oranges", secondaryLabel: "Jaffa" },
+      { id: "Bananas", primaryLabel: "Bananas", secondaryLabel: "Blue Java" },
+      { id: "Mangoes", primaryLabel: "Mangoes", secondaryLabel: "Alphonso" },
+      { id: "Kiwi", primaryLabel: "Kiwi", secondaryLabel: "Golden" },
+      { id: "Dragonfruit", primaryLabel: "Dragonfruit" },
+      { id: "Plum", primaryLabel: "Plum" },
+      { id: "Grapes", primaryLabel: "Grapes", secondaryLabel: "Mixed" },
+      { id: "Cherry", primaryLabel: "Cherry", secondaryLabel: "Black" }
+    ]
   },
 
-  render: Template
+  render: TransferList
+};
+
+// controlled component that calls the handleTransfer callback with data
+export const Controlled: StoryObj<typeof TransferList<TransferListItem>> = {
```

Comment:
I don't think this story works properly in storybook because you aren't rendering it with something that keeps track of the "controlled" state.